src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsEvoSuiteComponent.kt

```diff
@@ -34,55 +39,78 @@ class TestGenieSettingsComponent {
-            .addLabeledComponent(JBLabel("Select configuration id (null if left empty) "), configurationIdTextField, 5, false)
-            .addComponent(clientOnThreadCheckBox, 10)
-            .addComponent(minimizeCheckBox, 10)
-            .addComponent(junitCheckCheckBox, 10)
-            .addComponent(criterionSeparator, 15)
-            .addComponent(criterionLineCheckBox, 5)
-            .addComponent(criterionBranchCheckBox, 5)
-            .addComponent(criterionExceptionCheckBox, 5)
-            .addComponent(criterionWeakMutationCheckBox, 5)
-            .addComponent(criterionOutputCheckBox, 5)
-            .addComponent(criterionMethodCheckBox, 5)
-            .addComponent(criterionMethodNoExceptionCheckBox, 5)
-            .addComponent(criterionCBranchCheckBox, 5)
-            .addComponentFillVertically(JPanel(), 0)
-            .panel
+                .addComponent(JXTitledSeparator("General settings"))
+
+                //EvoSuite "input" options (e.g. text, number)
+                //Important settings like algorithm selection, seed selection
+                .addLabeledComponent(JBLabel("Select search algorithm"), algorithmSelector, 10, false)
+                .addLabeledComponent(JBLabel("Seed(random if left empty) "), seedTextField, 10, false)
+                .addLabeledComponent(JBLabel("Select configuration id (null if left empty) "), configurationIdTextField, 5, false)
+
+                //Checkboxes settings
+                .addComponent(sandboxCheckBox, 10)
+                .addComponent(assertionsCheckBox, 10)
+                .addComponent(clientOnThreadCheckBox, 10)
+                .addComponent(minimizeCheckBox, 10)
+                .addComponent(junitCheckCheckBox, 10)
+
+                //Criterion selection checkboxes
+                .addComponent(criterionSeparator, 15)
+                .addComponent(criterionLineCheckBox, 5)
+                .addComponent(criterionBranchCheckBox, 5)
+                .addComponent(criterionExceptionCheckBox, 5)
+                .addComponent(criterionWeakMutationCheckBox, 5)
+                .addComponent(criterionOutputCheckBox, 5)
+                .addComponent(criterionMethodCheckBox, 5)
+                .addComponent(criterionMethodNoExceptionCheckBox, 5)
+                .addComponent(criterionCBranchCheckBox, 5)
+                .addComponentFillVertically(JPanel(), 0)
+                .panel
+    }
 
+    /**
+     * Add stylistic additions to elements of EvoSuite settings panel (e.g. tooltips)
+     */
+    fun stylizePanel() {
+        //Dimensions adjustments
         algorithmSelector.setMinimumAndPreferredWidth(300)
+
+        //Tooltips
         configurationIdTextField.toolTipText = "Label that identifies the used configuration of EvoSuite. This is only done when running experiments."
         clientOnThreadCheckBox.toolTipText = "Run client process on same JVM of master in separate thread. To be used only for debugging purposes"
```

Comment:
It is nice that you extracted putting tooltips into a separate method. Looks much cleaner now